core/src/main/java/bisq/core/dao/state/unconfirmed/UnconfirmedBsqChangeOutputListService.java

```diff
@@ -0,0 +1,194 @@
+
+    /**
+     * Once a tx gets committed to out BSQ wallet we store the change output for allowing it to be spent in follow-up
+     * transactions.
+     */
+    public void onCommitTx(Transaction tx, TxType txType, Wallet wallet) {
+        // We remove all potential connected outputs from our inputs as they would have been spent.
+        removeConnectedOutputsOfInputsOfTx(tx);
+
+        int changeOutputIndex;
+        switch (txType) {
+            case UNDEFINED_TX_TYPE:
+            case UNVERIFIED:
+            case INVALID:
+            case GENESIS:
+                return;
+            case TRANSFER_BSQ:
+                changeOutputIndex = 1; // output 0 is receivers address
+                break;
+            case PAY_TRADE_FEE:
+                changeOutputIndex = 0;
+                break;
+            case PROPOSAL:
+                changeOutputIndex = 0;
+                break;
+            case COMPENSATION_REQUEST:
+            case REIMBURSEMENT_REQUEST:
+                changeOutputIndex = 0;
+                break;
+            case BLIND_VOTE:
+                changeOutputIndex = 1; // output 0 is stake
+                break;
+            case VOTE_REVEAL:
+                changeOutputIndex = 0;
+                break;
+            case LOCKUP:
+                changeOutputIndex = 1; // output 0 is lockup amount
+                break;
+            case UNLOCK:
+                // We don't allow to spend the unlocking funds as there is the lock time which need to pass,
+                // otherwise the funds get burned!
+                return;
+            case ASSET_LISTING_FEE:
+                changeOutputIndex = 0;
+                break;
+            case PROOF_OF_BURN:
+                changeOutputIndex = 0;
+                break;
+            default:
+                return;
+        }
+
+        // It can be that we don't have a BSQ and a BTC change output.
+        // If no BSQ change but a BTC change the index points to the BTC output and then
+        // we detect that it is not part of our wallet.
+        // If there is a BSQ change but no BTC change it has not effect as we ignore BTC outputs anyway.
+        // If both change outputs do not exist then we might point to an index outside
+        // of the list and we return at our scope check.
+
+        // If no BTC output (unlikely but
+        // possible) the index points to the BTC output and then we detect that it is not part of our wallet.
+        //
+        List<TransactionOutput> outputs = tx.getOutputs();
+        if (changeOutputIndex > outputs.size() - 1)
+            return;
+
+        TransactionOutput change = outputs.get(changeOutputIndex);
+        if (!change.isMine(wallet))
+            return;
+
+        UnconfirmedTxOutput txOutput = UnconfirmedTxOutput.fromTransactionOutput(change);
+        if (unconfirmedBsqChangeOutputList.containsTxOutput(txOutput))
+            return;
+
+        log.error("We add txOutput {}", txOutput);
```

Comment:
Why log an error?

Reply:
Was dev debugging... was remove in follow up commit